test/module_definition/parameter_test.py

```diff
@@ -37,3 +46,113 @@ def test_initial_kind_is_guessed():
     # then:
     assert parameter.is_input
     assert parameter.is_output
+
+
+def test_enrich_with_documentation_defaults():
+    # given:
+    parameter = Parameter('abc', 'def', SimpleType('int'))
+    parameter_documentation = create_parameter_documentation('abc')
+
+    # when:
+    parameter.enrich_with_documentation(parameter_documentation)
+
+    # then:
+    assert parameter.kind == ParameterKind.kind_in()
+    assert parameter.is_input
+    assert not parameter.is_output
+    assert parameter.has_simple_type
+    assert not parameter.has_pointer_type
+    assert not parameter.is_single_element
+    assert not parameter.has_fixed_length
+    assert parameter.fixed_length is None
+    assert not parameter.is_null_terminated
+    assert not parameter.has_length_descriptor
+    assert parameter.length_descriptor is None
+
+
+def test_enrich_with_documentation_sets_fixed_length():
+    # given:
+    parameter = Parameter('abc', 'def', PointerType('int', False, 1, False))
+    parameter_documentation = create_parameter_documentation('abc',
+                                                             fixed_length=4)
+
+    # when:
+    parameter.enrich_with_documentation(parameter_documentation)
+
+    # then:
+    assert parameter.kind == ParameterKind.kind_in_out()
+    assert parameter.fixed_length == 4
+
+
+def test_has_fixed_length_is_the_default_for_pointers():
+    # given:
+    parameter = Parameter('abc', 'def', PointerType('int', False, 1, False))
+    documentation = create_parameter_documentation('abc')
+
+    # when:
+    parameter.enrich_with_documentation(documentation)
+
+    # then:
+    assert not parameter.has_fixed_length
```

Comment:
The name doesn't match the check. Adapt the name to match what this test actually checks.